bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/SystemUserService.cs

```diff
@@ -50,13 +54,57 @@ public async Task<List<SystemUser>> GetAllSystemUserDTOsForChosenUser(int id, Ca
 
     public async Task<SystemUser?> PostNewSystemUserDescriptor(int partyId, SystemUserDescriptor newSystemUserDescriptor, CancellationToken cancellation = default)
     {
-        AuthorizedPartyExternal party = await _partyLookUpClient.GetPartyFromReporteeListIfExists(partyId);
-        string partyOrgNo = party.OrganizationNumber; 
-        return await _systemUserClient.PostNewSystemUserReal(partyOrgNo, newSystemUserDescriptor, cancellation);
+        AuthorizedPartyExternal party = await _partyLookUpClient.GetPartyFromReporteeListIfExists(partyId);        
+        string partyOrgNo = party.OrganizationNumber;
+        bool canDelegate = await UserDelegationCheckForReportee(partyId, newSystemUserDescriptor.SelectedSystemType!, cancellation);
+        if (canDelegate)
+        {
+            return await _systemUserClient.PostNewSystemUserReal(partyOrgNo, newSystemUserDescriptor, cancellation);
+        }
+
+        else return null;
+        
     }
 
     public async Task<bool> ChangeSystemUserProduct(string selectedSystemType, Guid id, CancellationToken cancellationToken = default)
     {
         return await _systemUserClient.ChangeSystemUserRealProduct(selectedSystemType, id, cancellationToken);
     }
+
+    public async Task<bool> UserDelegationCheckForReportee(int loggedInPartyId, string systemId ,CancellationToken cancellationToken = default)
```

Comment:
if loggedInPartyID is the partyID for the selected reportee just call it partyId.  If this is the userID for the user doing this call it userID

Reply:
okay